miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/RunRestController.java

```diff
@@ -264,9 +264,11 @@ public void addContainerByBarcode(@PathVariable Long runId, @RequestParam("barco
       throw new RestException("Multiple containers with this barcode.", Status.BAD_REQUEST);
     }
     SequencerPartitionContainer container = containers.iterator().next();
-    if (container.getPlatform().getId() != run.getSequencer().getPlatform().getId()) {
-      throw new RestException(String.format("This container is meant for %s but this run came from %s.",
-          container.getPlatform().getNameAndModel(), run.getSequencer().getPlatform().getNameAndModel()), Status.BAD_REQUEST);
+    if (!container.getModel().getPlatforms().stream()
+        .anyMatch(platform -> platform.getId() == run.getSequencer().getPlatform().getId())) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this negation and "anyMatch()" with "noneMatch()". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4034)